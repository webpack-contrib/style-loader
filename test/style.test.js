/* eslint-disable */
import path from 'path';
import webpack from './helpers/compiler';

describe('Loader', () => {
  test('Style', async () => {
    const config = {
      rules: [
        {
          test: /\.css$/,
          use: [
            { loader: path.resolve(__dirname, '../src') },
            {
              loader: 'css-loader',
              options: {}
            }
          ]
        }
      ]
    };

    const stats = await webpack('fixture.js', config);
    const { source } = stats.toJson().modules[3];

    expect(source).toMatchSnapshot();
  });
});
