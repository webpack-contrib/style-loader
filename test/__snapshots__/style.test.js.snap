// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Style 1`] = `
"// Style Loader
// Adds CSS to the DOM by adding a <style> tag
import CSS from \\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\";
import style from \\"!../../src/lib/index.js\\";


// CSS Exports
export * from \\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\";
// CSS
let css = CSS

// Convert CSS
if (typeof css === \\"string\\") {
  css = [[module.id, css, \\"\\"]];
}

// Loader Options
const options = {\\"hmr\\":true}



// Add Styles (DOM)
const styles = style(css, options);

// Hot Module Replacement
if (module.hot) {
  // When the styles change, update the <style> tags
  module.hot.accept(\\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\", () => {
    // TODO update to ESM
    let hot = require(\\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\");

    if (typeof hot === \\"string\\") {
      hot = [[module.id, hot, \\"\\"]];
    }

    style(hot);
  });

  // When the module is disposed, remove the <style> tags
  module.hot.dispose(() => style());
}"
`;
